### PR TITLE
feat(core): globalShortcut API

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

This PR introduces a global shortcut API. Run the API example to test it out :)

```javascript
import { registerShortcut, unregisterShortcut } from '@tauri-apps/api/globalShortcut'

registerShortcut('ctrl + shift + a', () => alert('tauri!')).catch(console.error)
registerShortcut('super + T', () => alert('it works'))

unregisterShortcut('super + T').catch(console.error)
```

* Tested on Linux, macOS, Windows
* This only works because of the [wry](https://github.com/tauri-apps/wry) migration :)

### CODE REVIEW

**Comment by ken0x0a** [comment]
\[Question (macOS)]

Are shortcuts, registered using this PR code, appears as MenuItem like this?

**Comment by lucasfernog** [comment]
We don't have menu support yet but I think we'd need to do it manually.

**Comment by ken0x0a** [comment]
It seems this is something related to do that for macOS.
https://github.com/ryanmcgrath/cacao/blob/master/src/macos/menu/item.rs

**Comment by lucasfernog** [comment]
Cool, we'll handle that when WRY supports menus.

File: .changes/shortcut-api.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"api": patch
+"tauri-api": patch
+"tauri": patch
```
Comment (nklayman) [inline .changes/shortcut-api.md]: Wouldn't this be a minor release since it adds a feature?

File: api/src/globalShortcut.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { promisified, transformCallback } from './tauri'
+
+/**
+ * register a global shortcut
+ * @param shortcut shortcut definition, modifiers and key separated by "+" e.g. Control+Q
```
Comment (nklayman) [inline api/src/globalShortcut.ts]: With the example you provide here, I think it would be better to use CommandOrControl so that it works the same on Mac. See [this](https://www.electronjs.org/docs/api/accelerator#platform-notice) for more details.

File: api/src/globalShortcut.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      cmd: 'register',
+      shortcut,
+      handler: transformCallback(handler)
+    }
+  })
+}
+
+/**
+ * unregister a global shortcut
+ * @param shortcut shortcut definition, modifiers and key separated by "+" e.g. Control+Q
```
Comment (nklayman) [inline api/src/globalShortcut.ts]: Same as above

File: tauri-api/Cargo.toml [hunk 4]
```diff
@@ -35,6 +35,7 @@ tauri-utils = { version = "0.5", path = "../tauri-utils" }
 clap = { version = "=3.0.0-beta.2", optional = true }
 notify-rust = { version = "4.2.2", optional = true }
 once_cell = "1.5.2"
+tauri-hotkey = { git = "https://github.com/tauri-apps/tauri-hotkey-rs", branch = "dev", optional = true }
```
Comment (nklayman) [inline tauri-api/Cargo.toml:38]: Shouldn't this get published to crates.io before we release? Do we want to do it before merge so we don't forget?

File: tauri/src/endpoints/global_shortcut.rs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  /// Unregister a global shortcut.
+  Unregister {
+    shortcut: String,
+    callback: String,
+    error: String,
+  },
+}
+
+impl Cmd {
+  #[allow(unused_variables)]
```
Comment (nklayman) [inline tauri/src/endpoints/global_shortcut.rs]: Why is this here?

**Review by nklayman — CHANGES_REQUESTED** [review]
Awesome feature to have! Just a few questions/suggestions.

File: tauri-api/Cargo.toml (hunk 4 above)
Comment (lucasfernog) [reply tauri-api/Cargo.toml:38]: we will publish it, but it's still in development so i'll just leave it as a git dep here. (yeah let's hope we don't forget it :joy: )

File: tauri/src/endpoints/global_shortcut.rs (hunk 5 above)
Comment (lucasfernog) [reply tauri/src/endpoints/global_shortcut.rs]: hmm I rewrote the method and it isn't needed anymore; thanks for finding that

File: api/src/globalShortcut.ts (hunk 2 above)
Comment (lucasfernog) [reply api/src/globalShortcut.ts]: CommandOrControl isn't supported yet, will be added later

File: tauri-api/Cargo.toml (hunk 4 above)
Comment (nklayman) [reply tauri-api/Cargo.toml:38]: We might want to consider making a list or adding it to that hackmd at least.

File: api/src/globalShortcut.ts (hunk 2 above)
Comment (nklayman) [reply api/src/globalShortcut.ts]: Oh ok. Maybe we could use `alt` instead since that is cross-platform and then we don't have to remember to change this later when we do add CommandOrControl.

Comment (lucasfernog) [reply api/src/globalShortcut.ts]: Ctrl works just fine on macOS though

File: tauri-api/Cargo.toml (hunk 4 above)
Comment (lucasfernog) [reply tauri-api/Cargo.toml:38]: yeah we should manage our TODOs better

File: api/src/globalShortcut.ts (hunk 2 above)
Comment (nklayman) [reply api/src/globalShortcut.ts]: It does work, but according to macworld.com:
> On a Mac you use the Command key where on a PC you would use Control (or Ctrl)

So event though it is a key, it's not what a user would expect it to be.

Comment (lucasfernog) [reply api/src/globalShortcut.ts]: there you go

Comment (nklayman) [reply api/src/globalShortcut.ts]: Sorry for nitpicking, but I could totally see someone copying that example and then opening an issue complaining that "Tauri keyboard shortcuts are broken reeeeee" :joy:.
